python/lsst/daf/butler/datastores/sasquatchDatastore.py

```diff
@@ -0,0 +1,885 @@
+            collection the bundle is stored in the butler.
+        datasetType : `str`
+            The dataset type name associated with this MetricMeasurementBundle
+        timestamp : `str` or `None`
+            The timestamp to be associated with the measurements in the ingress
+            database. If this value is None, timestamp will be set by the run
+            time or current time.
+        id : `UUID` or `None`
+            The UUID of the MetricMeasurementBundle within the butler. If None,
+            a new random UUID will be generated so that each record in
+            Sasquatch will have a unique value.
+        datasetIdentifier : `str`
+            A string which will be used in creating unique identifier tags.
+        identifierFields: `Mapping` or `None`
+            The keys and values in this mapping will be both added as fields
+            in the record, and used in creating a unique tag for the uploaded
+            dataset type. I.e. the timestamp, and the tag will be unique, and
+            each record will belong to one combination of such.
+        extraFields: `Mapping`
+            Extra mapping keys and values that will be added as fields to the
+            dispatched record.
+
+        Returns
+        -------
+        result : `Mapping` of `str` to `list`
+            A mapping of metric name of list of metric measurement records
+        status : `bool`
+            A status bool indicating if some records had to be skipped due to
+            a problem parsing the bundle.
+        """
+        if id is None:
+            id = uuid4()
+        sid = str(id)
+        meta: dict[str, Any] = dict()
+
+        # Add other associated common fields
+        meta["id"] = sid
+        meta["run"] = run
+        meta["dataset_type"] = datasetType
+
+        # Check to see if the bundle declares a reference package
+        self._handleReferencePackage(meta, bundle)
+
+        # Handle the various timestamps that could be associated with a record
+        self._handleTimes(meta, bundle, run)
+
+        # Always use the supplied timestamp if one was passed to use.
+        if timestamp is not None:
+            meta["timestamp"] = timestamp.timestamp()
+
+        self._handleIdentifier(meta, identifierFields, datasetIdentifier, bundle)
+
+        # Add in any other fields that were supplied to the function call.
+        if extraFields is not None:
+            meta.update(extraFields)
+
+        metricRecords: dict[str, list[Any]] = dict()
+
+        # Record if any records needed skipped
+        resultsTrimmed = False
+
+        # Look at each of the metrics in the bundle (name, values)
+        for metric, measurements in bundle.items():
+            # create a list which will contain the records for each measurement
+            # associated with metric
+            metricRecordList = metricRecords.setdefault(metric, list())
+
+            record: dict[str, Any] = dict()
```

Comment:
Is it more efficient to do this as:

```python
record = meta.copy()
```
rather than using `update()` immediately?